### PR TITLE
 #169643311  Users should be able to get notifications

### PR DESCRIPTION
**Pull request by nkalyesubula**
- #### What does this PR do?

- Adds functionality for getting user notifications

#### Description of Task to be completed?

- Given a user opted-in for notifications, he should be able to receive them every time a person he follows publishes a new article or when an article he favorited receives new reactions e.g comments

#### How should this be manually tested?

- clone repo, and then run npm run dev in your terminal

- Go to the postman and run this route :localhost:8080/api/v1/notifications

#### What are the relevant pivotal tracker stories?
[#169643311](https://www.pivotaltracker.com/story/show/169643311)
#### Screenshots (if appropriate)

image
image

image
image

### CODE REVIEW

File: src/controllers/notifications.controller.js [hunk 1]
```diff
@@ -31,5 +31,30 @@ class NotificationController {
...
+ * @param {object} req client request
+ * @param {object} res server response
+ * @returns {Object} server response object
+ * @param  {Function} next passes control to the next middleware
+ * @memberof NotificationController
+ */
+  static async getUserNotification(req, res) {
+    try {
+      const user = req.auth;
+      const userNotification = await models.AppNotification.findAll({
```
Comment (salviosage) [inline src/controllers/notifications.controller.js:51]: please consider if notification is new or the user already seen it.

File: src/controllers/notifications.controller.js [hunk 2]
```diff
@@ -31,5 +31,30 @@ class NotificationController {
...
+        where: { receiverId: user.id },
+      });
+      if (userNotification.length === 0) {
+        return res.status(404).json({ status: 404, message: 'You currently do not have notifications' });
+      }
+      return res.send({ notifications: userNotification });
+    } catch (error) {
```
Comment (salviosage) [inline src/controllers/notifications.controller.js:91]: consider also using util function while returning the response.

**Review by salviosage — COMMENTED** [review]
nice work, I have left some comment.

File: src/controllers/notifications.controller.js (hunk 1 above)
Comment (nkalyesubula) [reply src/controllers/notifications.controller.js:51]: Fixed

File: src/controllers/notifications.controller.js (hunk 2 above)
Comment (nkalyesubula) [reply src/controllers/notifications.controller.js:91]: Fixed

**Review by minega25 — APPROVED** [review]
LGTM

**Review by nshutijonathan — APPROVED** [review]

File: src/controllers/notifications.controller.js (hunk 1 above)
Comment (salviosage) [reply src/controllers/notifications.controller.js:51]: how I can't see it

**Review by salviosage — APPROVED** [review]
LGTM

**Pull request merged**